src/shared_modules/content_manager/src/actionOrchestrator.hpp

```diff
@@ -142,6 +124,71 @@ class ActionOrchestrator final
         m_spBaseContext->downloadedFileHash.clear();
     }
 
+    /**
+     * @brief Creates and triggers a new orchestration that updates the offset in the database.
+     *
+     * @param spUpdaterContext Updater context.
+     * @param offset New value of the offset.
+     */
+    void runOffsetUpdate(std::shared_ptr<UpdaterContext> spUpdaterContext, int offset) const
+    {
+        logDebug2(WM_CONTENTUPDATER, "Running '%s' offset update", m_spBaseContext->topicName.c_str());
+
+        if (0 > offset)
+        {
+            throw std::invalid_argument {"Invalid offset value: " + std::to_string(offset)};
+        }
+
+        spUpdaterContext->currentOffset = offset;
+
+        FactoryOffsetUpdater::create(m_spBaseContext->configData)->handleRequest(spUpdaterContext);
+    }
+
+    /**
+     * @brief Triggers a new orchestration that updates the content.
+     *
+     * @param spUpdaterContext Updater context.
+     * @param offset If zero, resets the current offset.
+     */
+    void runContentUpdate(std::shared_ptr<UpdaterContext> spUpdaterContext, int offset) const
+    {
+        logDebug2(WM_CONTENTUPDATER, "Running '%s' content update", m_spBaseContext->topicName.c_str());
+
+        // If the database exists, get the last offset
+        if (m_spBaseContext->spRocksDB)
+        {
+            spUpdaterContext->currentOffset = std::stoi(
+                m_spBaseContext->spRocksDB->getLastKeyValue(Components::Columns::CURRENT_OFFSET).second.ToString());
+        }
+
+        if (offset == 0)
+        {
+            spUpdaterContext->currentOffset = 0;
+        }
+
```

Comment:
Questions:
 - `spRocksDB` being null is valid? There is no need to throw an exception? it is also used on line 186 to update the hash, is that optional?
 - `offset` is only used if the value is zero. Is that correct? Does it make sense to turn it into a boolean? `resetOffset`?

Reply:
> spRocksDB being null is valid? There is no need to throw an exception? it is also used on line 186 to update the hash, is that optional?

`spRocksDB` is set in the Execution Context stage only when `databasePath` is defined (otherwise, it remains null). AFAIK, we don't have any use case where we don't use the RocksDB database.

Maybe @Damian-Mangold can help me answer your questions :sweat_smile:

Reply:
> offset is only used if the value is zero. Is that correct? Does it make sense to turn it into a boolean? resetOffset?

That's better. Changed.

Reply:
> spRocksDB is set in the Execution Context stage only when databasePath is defined (otherwise, it remains null). AFAIK, we don't have any use case where we don't use the RocksDB database.

Ok, then it seems valid for it to be null. No problem here.